Can't build github.com/docker/docker/pkg/system on freebsd

**Description**

This is an issue that was reported in fsouza/go-dockerclient#878: when trying to build go-dockerclient on freebsd, users see an error where `github.com/docker/docker/pkg/system` fails to build (it's a transient dependency of go-dockerclient).

Not sure if yall care about it or not, but I figured I could open an issue asking whether you'd take a PR before sending one or simply vendoring & fixing on my side.

Thank you very much!

**Steps to reproduce the issue:**

Clone go-dockerclient and try to build it with `GOOS=freebsd`:

```
% git clone --depth=1 https://github.com/fsouza/go-dockerclient.git
% cd go-dockerclient
% GOOS=freebsd go build 
```

**Describe the results you received:**

Compilation fails with the following error message:

```
# github.com/docker/docker/pkg/system
../../go/pkg/mod/github.com/docker/docker@v20.10.8+incompatible/pkg/system/mknod.go:12:19: cannot use dev (type int) as type uint64 in argument to unix.Mknod
```

**Describe the results you expected:**

Successful build.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
(paste your output here)
```

**Output of `docker info`:**

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

--- a/pkg/system/mknod.go
+++ b/pkg/system/mknod.go
@@ -7,12 +7,6 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-// Mknod creates a filesystem node (file, device special file or named pipe) named path
-// with attributes specified by mode and dev.
-func Mknod(path string, mode uint32, dev int) error {
-	return unix.Mknod(path, mode, dev)
-}
-
 // Mkdev is used to build the value of linux devices (in /dev/) which specifies major
 // and minor number of the newly created device special file.
 // Linux device nodes are a bit weird due to backwards compat with 16 bit device nodes.

--- a/pkg/system/mknod_freebsd.go
+++ b/pkg/system/mknod_freebsd.go
@@ -1,0 +1,14 @@
+//go:build freebsd
+// +build freebsd
+
+package system // import "github.com/docker/docker/pkg/system"
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+// Mknod creates a filesystem node (file, device special file or named pipe) named path
+// with attributes specified by mode and dev.
+func Mknod(path string, mode uint32, dev int) error {
+	return unix.Mknod(path, mode, uint64(dev))
+}

--- a/pkg/system/mknod_linux.go
+++ b/pkg/system/mknod_linux.go
@@ -1,0 +1,11 @@
+package system // import "github.com/docker/docker/pkg/system"
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+// Mknod creates a filesystem node (file, device special file or named pipe) named path
+// with attributes specified by mode and dev.
+func Mknod(path string, mode uint32, dev int) error {
+	return unix.Mknod(path, mode, dev)
+}

